### PR TITLE
Removing extraneous space inserted during refactoring commit 894484a.…

### PR DESCRIPTION
**Pull request by mike-kingsbury**
…  Code was moved to a new function, and a space was added to the split.  bz#2126155

### All Pull Requests:

Check all that apply:

* [ x ] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [ x ] Is this PR to correct an issue?
* [ ] Is this PR an enhancement?

### Complete Description of Additions/Changes:

Reverting a change that introduced a space into a .split(',').  The space broke the functionality of the Insights client when using NO_PROXY with multiple items.

### CODE REVIEW

**Comment by xiangce** [comment]
I'm just wondering if we could add a test case for this? It seems an issue that can be detected by a sufficient test.

**Comment by mike-kingsbury** [comment]
> I'm just wondering if we could add a test case for this? It seems an issue that can be detected by a sufficient test.

Give me a couple of minutes, you're correct it should be testable.

**Comment by xiangce** [comment]
@mike-kingsbury - Thanks for the update. It looks good to me.  @ahitacat, please have a look, if it is also good for you, I think we can merge and release today.

**Comment by ahitacat** [comment]
In that case it would fail with `no_proxy` values separated by comma **and** space. Like the one is given in the [bug report](https://bugzilla.redhat.com/show_bug.cgi?id=2065233) This no_proxy function get the the value from env or from rhsm. So let's try to support bot of them.

File: insights/client/connection.py [hunk 1]
```diff
@@ -276,7 +276,7 @@ def get_proxy(self, proxy_info, no_proxy_info, environment):
         if no_proxy_info:
             insights_service_host = urlparse(self.base_url).hostname
             logger.debug('Found NO_PROXY set. Checking NO_PROXY %s against base URL %s.', no_proxy_info, insights_service_host)
-            for no_proxy_host in no_proxy_info.split(', '):
+            for no_proxy_host in no_proxy_info.split(','):
```
Comment (ahitacat) [inline insights/client/connection.py:279]:
```suggestion
            for no_proxy_host in no_proxy_info.split(','):
                no_proxy_host.strip()
```
To remove any white spaces added while listing the values.

File: insights/tests/client/connection/test_init_session.py [hunk 2]
```diff
@@ -83,6 +83,14 @@ def test_get_no_proxy_env():
         connection.get_proxies()
     assert connection.proxies is None
 
+    with patch.dict(os_environ, {"HTTPS_PROXY": "env.proxy.example.com", "NO_PROXY": "redhat.com,example.com"}, clear=True):
```
Comment (ahitacat) [inline insights/tests/client/connection/test_init_session.py:86]:
```suggestion
    with patch.dict(os_environ, {"HTTPS_PROXY": "env.proxy.example.com", "NO_PROXY": "example.com,redhat.com"}, clear=True):
```
The problem appears when getting values after a comma.

**Review by ahitacat — CHANGES_REQUESTED** [review]
Also it could be useful to add a check under `test_get_no_proxy_rhsm` that tests when `no_proxy` has values separated by comma and space.

~~~
    with patch.dict(os_environ, {"HTTPS_PROXY": "env.proxy.example.com", "NO_PROXY": "example.com, redhat.com"}, clear=True):
        connection.get_proxies()
    assert connection.proxies == {'https': 'env.proxy.example.com'}
 ~~~

File: insights/client/connection.py (hunk 1 above)
Comment (mike-kingsbury) [reply insights/client/connection.py]: Almost....   .strip() doesn't modify the string, it returns a new string.  Implemented as such.

File: insights/tests/client/connection/test_init_session.py (hunk 2 above)
Comment (mike-kingsbury) [reply insights/tests/client/connection/test_init_session.py:86]: I've added a series of new test cases including spaces and different ordering of the domain names.

Comment (ahitacat) [reply insights/tests/client/connection/test_init_session.py:86]: This looks great :smile:

**Review by ahitacat — COMMENTED** [review]
This changes looks great and it can be merged. Thanks @mike-kingsbury for open and fix the bug.

**Review by ahitacat — APPROVED** [review]

**Pull request merged**